### PR TITLE
Fix deposit form issues

### PR DESCRIPTION
**Pull request by LeoVS09**

### CODE REVIEW

File: app/earn/[...vault]/form/form-button/form-button.tsx [hunk 1]
```diff
@@ -66,38 +75,63 @@ export default FormButton
...
+
+  const { formAction } = useVaultContext()
+  const { walletBalanceBN, vaultBalanceBN } = useAppSelector(state => state.vaultUser)
+  const haveEnoughAssets = formAction === FormAction.DEPOSIT ? inputValue <= BigInt(walletBalanceBN) : inputValue <= BigInt(vaultBalanceBN)
+
+  const canSubmit = walletAvailable && haveInputValue && haveEnoughAssets
 
   const submit = useCallback(async (formAction: FormAction) => {
+    if (!canSubmit) {
+      toast('Looks like something wrong, try refresh the page', {
```
Comment (solovev) [inline app/earn/[...vault]/form/form-button/form-button.tsx:96]:
```suggestion
      toast('Looks like something is wrong, try refreshing the page', {
```

File: app/earn/[...vault]/form/form-button/form-button.tsx [hunk 2]
```diff
@@ -66,38 +75,63 @@ export default FormButton
...
-        void refetechVaultUserData!()
+        void refetechVaultUserData()
 
         // Reset form input
         onValueChange(0n)
       }
     }
     catch (error) {
       console.error('Error during submit', error)
+      toast('An error occurred, please try refresh the page', {
```
Comment (solovev) [inline app/earn/[...vault]/form/form-button/form-button.tsx:119]:
```suggestion
      toast('An error occurred, please try refreshing the page', {
```

File: app/earn/[...vault]/form/form-button/form-button.tsx [hunk 3]
```diff
@@ -43,10 +45,14 @@ const FormButton: React.FC<Props> = ({ vaultChain, isLoading, disabled, ...restP
     void submit(formAction)
   }, [insured, setInsured, status, isOnDifferentChain, formAction, connect, setCurrentChain, submit, vaultChain.id])
 
+  const shouldBeAblToSubmit = status === WalletStatus.CONNECTED
```
Comment (solovev) [inline app/earn/[...vault]/form/form-button/form-button.tsx:48]:
```suggestion
  const shouldBeAbleToSubmit = status === WalletStatus.CONNECTED
```

Comment (solovev) [reply app/earn/[...vault]/form/form-button/form-button.tsx:48]: or even
```suggestion
  const isAbleToSubmit = status === WalletStatus.CONNECTED
```

**Review by solovev — APPROVED** [review]

File: app/earn/[...vault]/form/form-button/form-button.tsx (hunk 3 above)
Comment (LeoVS09) [reply app/earn/[...vault]/form/form-button/form-button.tsx:48]: This not check if user able to submit, rather check should we check "is he able to submit". So I will keep it as "shouldBeAbleToSubmit"

**Pull request merged**